grype/vulnerability_matcher.go

```diff
@@ -381,6 +303,43 @@ func (m *VulnerabilityMatcher) normalizeByCVE(match match.Match) match.Match {
 	return match
 }
 
+// ignoreRulesByLocation implements match.IgnoreFilter to filter each matching
+// package that overlaps by location and have the same vulnerability ID (CVE)
+type ignoreRulesByLocation struct {
+	locationToIgnoreRules map[string][]match.IgnoreRule
+}
+
+func (i ignoreRulesByLocation) IgnoreMatch(m match.Match) []match.IgnoreRule {
+	for _, l := range m.Package.Locations.ToSlice() {
+		for _, rule := range i.locationToIgnoreRules[l.RealPath] {
+			if rule.Vulnerability == m.Vulnerability.ID {
+				return []match.IgnoreRule{rule}
+			}
+			for _, relatedVulnerability := range m.Vulnerability.RelatedVulnerabilities {
+				if rule.Vulnerability == relatedVulnerability.ID {
+					return []match.IgnoreRule{rule}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ignoreMatchFilter creates an ignore filter based on the provided IgnoredMatches to filter out "the same"
+// vulnerabilities reported by other matchers based on overlapping file locations
+func ignoredMatchFilter(ignores []match.IgnoredMatch) match.IgnoreFilter {
+	out := ignoreRulesByLocation{locationToIgnoreRules: map[string][]match.IgnoreRule{}}
+	for _, ignore := range ignores {
+		// TODO should this be syftPkg.FileOwner interface or similar?
+		if m, ok := ignore.Package.Metadata.(pkg.ApkMetadata); ok {
```

Comment:
this seems to be part of the migrated apk matcher code, should this have also been moved? or is this intended to be more generic?

Reply:
from an offline explanation this is left over from the migration but is still needed since the ignore rules need to be applied to the ignored matches